extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusAuthenticationProviderModule.java

```diff
@@ -57,6 +62,24 @@ public RadiusAuthenticationProviderModule(AuthenticationProvider authProvider)
 
         // Get local environment
         this.environment = new LocalEnvironment();
+        
+        // Check for MD4 requirement
+        RadiusAuthenticationProtocol authProtocol = environment.getProperty(RadiusGuacamoleProperties.RADIUS_AUTH_PROTOCOL);
+        RadiusAuthenticationProtocol innerProtocol = environment.getProperty(RadiusGuacamoleProperties.RADIUS_EAP_TTLS_INNER_PROTOCOL);
+        if ((authProtocol != null 
+                    && (authProtocol == RadiusAuthenticationProtocol.MSCHAPv1 
+                    || authProtocol == RadiusAuthenticationProtocol.MSCHAPv2)) 
+                || (innerProtocol != null 
+                    && (innerProtocol == RadiusAuthenticationProtocol.MSCHAPv1 
+                    || innerProtocol == RadiusAuthenticationProtocol.MSCHAPv2))) {
```

Comment:
You might be able to clean up this logic if you remove the `null` checks, which are unnecessary in this case. The `==` comparisons will just fail if `authProtocol` or `innerProtocol` are `null`.

Reply:
Cleaned up.